bootstrap/_bootstrap_common.py

```
@@ -0,0 +1,257 @@
+
+def describe_table_columns(
+    db_conf: Dict[str, str], table_name: str
+) -> List[Dict[str, str]]:
+    safe_table = table_name.replace("'", "''")
+    output = psql_query(
+        db_conf,
+        "SELECT column_name, data_type, is_nullable, "
+        "COALESCE(column_default, '') "
+        "FROM information_schema.columns "
+        f"WHERE table_schema='public' AND table_name='{safe_table}' "
+        "ORDER BY ordinal_position;",
```

Ruff (0.15.10)

[error] Possible SQL injection vector through string-based query construction

(S608)